src/olympia/devhub/forms.py

```diff
@@ -1433,4 +1434,8 @@ def clean_origin(self):
             or parsed.fragment
         ):
             raise forms.ValidationError(error_message)
+        if DeniedInstallOrigin.find_denied_origins([value]):
```

Comment:
The check for the validation is for regular add-ons that specify an install origin in their manifests. We aren't currently enforcing the presence of that key, but the idea was that we would eventually do that in some circumstances - that's the Distribution Control project.

Site permissions also need an install origin to be specified, so it's re-using the same mechanism, but because they are generated and bypass validation, the check needs to be added here.